lang/LangPrimSource/SC_Ableton_Link.cpp

```diff
@@ -0,0 +1,283 @@
+#ifdef SC_ABLETON_LINK
+
+#include "PyrKernel.h"
+#include "PyrSched.h"
+#include "GC.h"
+#include "PyrPrimitive.h"
+#include "PyrSymbol.h"
+
+#include "SCBase.h"
+#include "SC_Clock.hpp"
+
+#include <ableton/Link.hpp>
+
+static std::chrono::microseconds linkTimeOfInitialization;
```

Comment:
Two questions about this:

1. If it's a static member variable, will it be initialized during process startup, or only when using LinkClock?

   At one point, while working on this, I did try to make it a member variable (though, not static). Initializing the variable in the LinkClock constructor broke sync. The variable must be initialized at process startup, or none of this works.

2. If it's a member variable, then https://github.com/supercollider/supercollider/blob/04bb5433b7ae369b6439524ccc26876d096a2dee/lang/LangPrimSource/SC_Ableton_Link.cpp#L15-L18 and https://github.com/supercollider/supercollider/blob/04bb5433b7ae369b6439524ccc26876d096a2dee/lang/LangPrimSource/PyrSched.cpp#L243-L245 need to be moved somewhere else. Where?

Note that this variable is parallel to `hrTimeOfInitialization`, declared at https://github.com/supercollider/supercollider/blob/04bb5433b7ae369b6439524ccc26876d096a2dee/lang/LangPrimSource/PyrSched.cpp#L234 -- this goes back to James McCartney's original TempoClock implementation. He is not likely to use global variables casually. Monkeying around with this feels risky to me (and, even if it's not that risky, it's outside of my C++ skill set, so I have to ask for more detailed advice).

Reply:
> 1. If it's a static member variable, will it be initialized during process startup, or only when using LinkClock?

I'm not sure I understand... we want this "initialization" to happen in `schedInit` right? That seems right to me and nothing needs to change there, it's just a matter of where the variable is declared.

> 2.  ...need to be moved somewhere else. Where?

```cpp
/* SC_Ableton_Link.cpp */

class LinkClock {
    /* definition */
    static std::chrono::hrc::time_point initializationTime;
};

void initLink()
{
    LinkClock::initializationTime = ...;
}
```

> He is not likely to use global variables casually

This code was written at a different time with very different features available in C++, and because of that, with a very different engineering style from what is considered best practice in C++ today. If you have a real argument for why it should be kept that way, I would hear it, but this just sounds like superstition.

Reply:
You might consider making `initializationTime` a private static member, in which case `initLink` could be public static method of the `LinkClock` class which sets the value.

Then the call `initLink()` becomes `LinkClock::initLink()` or perhaps just `LinkClock::Init()`.

Reply:
Can I kindly request somebody else to make this change? As I said, this is way outside of what I understand in C++.

Reply:
FYI @brianlheim 

Here is a pass at the static member stuff:

https://github.com/jamshark70/supercollider/pull/3

It requires a bunch of things shift around.